src/index.js

```diff
@@ -29,361 +25,450 @@ class IPLDResolver {
     // Object with current list of active resolvers
     this.resolvers = {}
 
-    // API entry point
-    this.support = {}
-
-    // Adds support for an IPLD format
-    this.support.add = (multicodec, resolver, util) => {
-      if (this.resolvers[multicodec]) {
-        throw new Error('Resolver already exists for codec "' + multicodec + '"')
-      }
-
-      this.resolvers[multicodec] = {
-        resolver: resolver,
-        util: util
-      }
-    }
-
-    this.support.load = options.loadFormat || ((codec, callback) => {
-      callback(new Error(`No resolver found for codec "${codec}"`))
-    })
-
-    this.support.rm = (multicodec) => {
-      if (this.resolvers[multicodec]) {
-        delete this.resolvers[multicodec]
+    if (options.loadFormat === undefined) {
+      this.loadFormat = async (codec) => {
+        const codecName = multicodec.print[codec]
+        throw new Error(`No resolver found for codec "${codecName}"`)
       }
+    } else {
+      this.loadFormat = options.loadFormat
     }
 
     // Enable all supplied formats
     for (const format of options.formats) {
-      const { resolver, util } = format
-      const multicodec = resolver.multicodec
-      this.support.add(multicodec, resolver, util)
+      this.addFormat(format)
     }
   }
 
-  get (cid, path, options, callback) {
-    if (typeof path === 'function') {
-      callback = path
-      path = undefined
+  /**
+   * Add support for an IPLD Format.
+   *
+   * @param {Object} format - The implementation of an IPLD Format.
+   * @returns {void}
+   */
+  addFormat (format) {
+    // IPLD Formats are using strings instead of constants for the multicodec
+    const codecBuffer = multicodec.getCodeVarint(format.resolver.multicodec)
+    const codec = multicodec.getCode(codecBuffer)
+    if (this.resolvers[codec]) {
+      const codecName = multicodec.print[codec]
+      throw new Error(`Resolver already exists for codec "${codecName}"`)
```

Comment:
I wonder how often you really want to replace an existing format. I see it as a gatekeeper to prevent hard to debug bugs, where accidentally some format is replaced.